ui/src/Components/EditProfile.jsx

```diff
@@ -0,0 +1,266 @@
+import React from 'react';
+import { Button, Col, Form, FormGroup, Label, Input} from 'reactstrap';
+import NavBar from './NavBar';
+
+//Uncomment/Comment based on env
+// const URL = "http://localhost:3001/dev/editProfile";
+const URL = "https://bixe448nsa.execute-api.us-west-1.amazonaws.com/dev/editProfile";
+
+class CreateProfile extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            profile: null,
+            props: props,
+            firstName: '',
+            lastName: '',
+            email: '',
+            number: '',
+            profile_id: '',
+            ref1Name: '',
+            ref1Email: '',
+            ref2Name: '',
+            ref2Email: '',
+            govIdFlag: false,
+            govId: null,
+            toolPicFlag: false,
+            toolPic: null,
+        };
+
+        this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
+        this.handleLastNameChange = this.handleLastNameChange.bind(this);
+        this.handleEmailChange = this.handleEmailChange.bind(this);
+        this.handleNumChange = this.handleNumChange.bind(this);
+        this.handleRef1NameChange = this.handleRef1NameChange.bind(this);
+        this.handleRef1EmailChange = this.handleRef1EmailChange.bind(this);
+        this.handleRef2NameChange = this.handleRef2NameChange.bind(this);
+        this.handleRef1EmailChange = this.handleRef2EmailChange.bind(this);
+        this.handleLicenseUpload = this.handleLicenseUpload.bind(this);
+        this.handleToolUpload = this.handleToolUpload.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+
+    }
+
+    // function to get new information 
+    async componentDidMount()
+    {
+      const url = `https://bixe448nsa.execute-api.us-west-1.amazonaws.com/dev/viewProfile?id=${this.state.props.match.params.id}`;
+      const resp = await fetch(url);
+      const data = await resp.json();
+      console.log(data);
+      this.setState({profile: data, loading: false});
+
+      // assign all required form variable from now
+      this.state.firstName = this.state.profile.result.first_name;
+      this.state.lastName = this.state.profile.result.last_name;
+      this.state.number = this.state.profile.result.contact_num;
+    }
+
+    // functions to handle method update
+    handleFirstNameChange(event) {
+        this.setState({firstName: event.target.value});
+    }
+
+    handleLastNameChange(event) {
+            this.setState({lastName: event.target.value});
+    }
+
+    handleEmailChange(event) {
+        this.setState({email: event.target.value});
+    }
+
+    handleNumChange(event) {
+        this.setState({number: event.target.value});
+    }
+
+    handleRef1NameChange(event) {
+        this.setState({ref1Name: event.target.value});
+    }
+
+    handleRef1EmailChange(event) {
+        this.setState({ref1Email: event.target.value});
+    }
+
+    handleRef2NameChange(event) {
```

Comment:
I wonder if we could refactor all these methods a bit to take in two args, event and attribute name. That way it would look something like below, and we can delete a ton of this class

```
let handleChange = (event, stateVar) => {
this.setState({stateVar: event.target.value)
}

Reply:
Good call out - I simply took inspo from Gurleens changes Sprint 2. Should have been more aware. 😬 
Thoughts on making this part of our refactor efforts next sprint? This will need to be done on the createProfile component AND we need to add more fields (which is all work being done next sprint) 
I don't mind doing it now, but it might be worthwhile having 1 refactor ticket for this subset of changes.

Reply:
Sounds good man, let's push it to next sprint